altair/vegalite/v5/schema/_typing.py

```diff
@@ -86,6 +119,85 @@ def func(
     ): ...
 """
 
+
+class Value(TypedDict, Generic[T]):
+    """
+    A `Generic`_ single item ``dict``.
+
+    Parameters
+    ----------
+    value: T
+        Wrapped value.
+
+    .. _Generic:
+        https://typing.readthedocs.io/en/latest/spec/generics.html#generics
+
+    """
+
+    value: T
+
+
+ColorHex = Annotated[
+    LiteralString,
+    re.compile(r"#[0-9a-f]{2}[0-9a-f]{2}[0-9a-f]{2}([0-9a-f]{2})?", re.IGNORECASE),
+]
+"""
+A `hexadecimal`_ color code.
+
+Corresponds to the ``json-schema`` string format:
+
+    {"format": "color-hex", "type": "string"}
+
+Examples
+--------
+:
+
+    "#f0f8ff"
+    "#7fffd4"
+    "#000000"
+    "#0000FF"
+    "#0000ff80"
+
+.. _hexadecimal:
+    https://www.w3schools.com/html/html_colors_hex.asp
+"""
+
+
+def is_color_hex(obj: Any) -> TypeIs[ColorHex]:
+    """Return ``True`` if the object is a hexadecimal color code."""
+    # NOTE: Extracts compiled pattern from metadata,
+    # to avoid defining  in multiple places.
+    it = iter(get_args(ColorHex))
+    next(it)
+    pattern: re.Pattern[str] = next(it)
+    return bool(pattern.fullmatch(obj))
+
+
+class RowCol(TypedDict, Generic[T], total=False):
+    """
+    A `Generic`_ two-item ``dict``.
+
+    Parameters
+    ----------
+    column: T
+    row: T
+
+    .. _Generic:
+        https://typing.readthedocs.io/en/latest/spec/generics.html#generics
+
+    """
+
+    column: T
+    row: T
+
+
+class PaddingKwds(TypedDict, total=False):
+    bottom: float
+    left: float
+    right: float
+    top: float
```

Comment:
I can define these in `_config.py` instead if there's interest?

Tried to wrap things up since this PR has gotten quite large

Reply:
In [`aaffebd` (#3536)](https://github.com/vega/altair/pull/3536/commits/aaffebd2b2d037024a2f4af1e6d942bdd9e24174) I settled on including them in `_config.__all__`